inspirehep/modules/records/serializers/fields_export.py

```diff
@@ -0,0 +1,337 @@
+        date_choice = thesis_date or publication_year or imprint_date
+    else:
+        date_choice = publication_year or thesis_date or imprint_date
+
+    if date_choice:
+        return PartialDate.parse(unicode(date_choice))
+
+
+# Functions below describe where the non-obvious data is located:
+#
+# Args:
+#   data (dict): data partially serialized by the means of defined marshmallow schema only
+#   doc_type (str): BibTeX document type, as reported by `bibtex_document_type`
+
+extractor = make_extractor()
+
+
+@extractor('author')
+def get_author(data, doc_type):
+    """
+    Note:
+        Only used to generate author field if corporate_author is the author.
+    """
+    if 'corporate_author' in data:
+        return ' and '.join('{{{}}}'.format(author) for author in data['corporate_author'])
+
+
+@extractor('journal')
+def get_journal(data, doc_type):
+    return get_value(data, 'publication_info[0].journal_title')
+
+
+@extractor('volume')
+def get_volume(data, doc_type):
+    publication_volume = get_value(data, 'publication_info[0].journal_volume')
+    bookseries_volume = get_value(data, 'book_series[0].volume')
+    return publication_volume or bookseries_volume
+
+
+@extractor('year')
+def get_year(data, doc_type):
+    date = get_date(data, doc_type)
+    if date:
+        return date.year
+
+
+@extractor('month')
+def get_month(data, doc_type):
+    date = get_date(data, doc_type)
+    if date:
+        return date.month
+
+
+@extractor('number')
+def get_number(data, doc_type):
+    return get_value(data, 'publication_info[0].journal_issue')
+
+
+@extractor('pages')
+def get_pages(data, doc_type):
+    return get_page_artid(data, '--')
+
+
+@extractor('primaryClass')
+def get_primary_class(data, doc_type):
+    return get_value(data, 'arxiv_eprints[0].categories[0]')
+
+
+@extractor('eprint')
+def get_eprint(data, doc_type):
+    return get_value(data, 'arxiv_eprints[0].value')
+
+
+@extractor('archivePrefix')
+def get_arxiv_prefix(data, doc_type):
+    if get_eprint(data, doc_type):
+        return "arXiv"
+
+
+@extractor('SLACcitation')
+def get_slac_citation(data, doc_type):
+    def format_field(citation):
+        return "%%CITATION = {};%%".format(citation).upper()
+
+    doi = get_value(data, 'dois[0].value')
+    eprint = get_value(data, 'arxiv_eprints[0].value')
+    report_number = get_value(data, 'report_numbers[0].value')
+
+    if doi:
+        return format_field('DOI:' + doi)
+    elif eprint:
+        return format_field('ARXIV:' + eprint if is_arxiv_post_2007(eprint) else eprint)
+    elif report_number:
+        return format_field(report_number)
+
+
+@extractor('school')
+def get_school(data, doc_type):
+    schools = [school['name'] for school in get_value(data, 'thesis_info.institutions')]
+    if schools:
+        return ', '.join(schools)
+
+
+@extractor('address')
+def get_address(data, doc_type):
+    pubinfo_city = get_value(data, 'publication_info[0].conference_record.address[0].cities[0]')
```

Comment:
this info is not present in a Literature record, so you need to resolve some references for this to work. Are you doing that?

Reply:
Yes, in [inspirehep/modules/records/serializers/schemas/base.py#L42](https://github.com/szymonlopaciuk/inspire-next/blob/bibtex_rewrite/inspirehep/modules/records/serializers/schemas/base.py#L42) I replace_refs, I understand that's the correct thing to do?

Reply:
:+1: I overlooked it.